openedx/adg/lms/applications/tests/test_models.py

```diff
@@ -0,0 +1,82 @@
+"""
+Tests for all the models in applications app.
+"""
+
+import pytest
+
+from openedx.adg.lms.applications.models import ApplicationHub
+from student.tests.factories import UserFactory
+from .constants import USERNAME, EMAIL, PASSWORD
+
+
+@pytest.mark.django_db
+@pytest.fixture()
+def user_for_testing():
+    """
+    Create a test user
+
+    Returns:
+        User object
+
+    """
+    user = UserFactory(username=USERNAME, email=EMAIL, password=PASSWORD)
+    return user
+
+
+@pytest.mark.django_db
+@pytest.fixture()
+def application_hub_for_testing(user_for_testing):
+    """
+    Create an ApplicationHub object for the specified test user.
+
+    Args:
+        user_for_testing: User object
+
+    Returns:
+        ApplicationHub object
+
+    """
+    return ApplicationHub.objects.create(user=user_for_testing)
+
+
+@pytest.mark.django_db
+def test_set_is_prerequisite_courses_passed(user_for_testing, application_hub_for_testing):
+    """
+    Test if the is_prerequisite_courses_passed is being set correctly by the model method.
+
+    Args:
+        user_for_testing: User object
+        application_hub: ApplicationHub object
+
+    Returns:
+        Boolean
+
+    """
+    application_hub_for_testing.set_is_prerequisite_courses_passed()
+    assert ApplicationHub.objects.get(user=user_for_testing).is_prerequisite_courses_passed is True
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("attributes_set,expected", [
+    ([], 0),
+    (['is_prerequisite_courses_passed'], 1),
+    (['is_written_application_completed'], 1),
+    (['is_prerequisite_courses_passed', 'is_written_application_completed'], 2)
+])
+def test_number_of_objectives_completed(attributes_set, expected, application_hub_for_testing):
+    """
+    Test if the `number_of_objectives` property is working as expected for all possible cases
+
+    Args:
+         attributes_set: list; yhe model attributes to be set
+         expected: int; the expected value that the property should have
+         application_hub: ApplicationHub model
+
+    Returns:
+        Boolean
+
+    """
```

Comment:
Please have a look into this https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Types of arguments is not documented properly also what is "yhe"? I think its "The"?